src/main/java/com/manywho/services/dummy/account/repositories/AccountRepository.java

```diff
@@ -18,44 +20,144 @@ public AccountRepository(Sql2o sql2o) {
+        StringBuilder stringBuilder = new StringBuilder("SELECT * FROM account ");
+
+        if (filter != null && filter.getWhere() != null && filter.getWhere().size() != 0) {
+
+            if (filter.getWhere().size() > 1) {
+                throw new ServiceProblemException(400, "Currently we only support filtering on one value, multiple values have been are provided.");
+            }
+
+            String value = filter.getWhere().get(0).getContentValue();
+            String columnName = filter.getWhere().get(0).getColumnName();
+            CriteriaType criteriaType = filter.getWhere().get(0).getCriteriaType();
+
+            stringBuilder.append(" WHERE ");
+            stringBuilder.append(columnName);
+
+            if (columnName.equals("id")) {
+                switch (criteriaType) {
+                    case GreaterThan:
+                        stringBuilder.append(" > ");
+                        stringBuilder.append(value);
+                        break;
+                    case GreaterThanOrEqual:
+                        stringBuilder.append(" >= ");
+                        stringBuilder.append(value);
+                        break;
+                    case LessThan:
+                        stringBuilder.append(" < ");
+                        stringBuilder.append(value);
+                        break;
+                    case LessThanOrEqual:
+                        stringBuilder.append(" <= ");
+                        stringBuilder.append(value);
+                        break;
+                    case Equal:
+                        stringBuilder.append(" = ");
+                        stringBuilder.append(value);
+                        break;
+                    case NotEqual:
+                        stringBuilder.append(" != ");
+                        stringBuilder.append(value);
+                        break;
+                    case IsEmpty:
+                        stringBuilder.append(" > ");
+                        stringBuilder.append(value);
+                        stringBuilder = (Boolean.valueOf(filter.getWhere().get(0).getContentValue())) ?
+                                stringBuilder.append(" is null ") :
+                                stringBuilder.append(" is not null");
+                        break;
+                    default:
+                        throw new ServiceProblemException(400, "Integer CriteriaType only supports EQUAL, NOT_EQUAL, GREATER_THAN, GREATER_THAN_OR_EQUAL, LESS_THAN, LESS_THAN_OR_EQUAL and IS_EMPTY operations");
+                }
+            } else if (columnName.equals("company") || columnName.equals("name") || columnName.equals("stateid")) {
+                switch (criteriaType) {
+                    case StartsWith:
+                        stringBuilder.append(" LIKE \'");
+                        stringBuilder.append(value);
+                        stringBuilder.append("%\'");
+                        break;
```

Comment:
The three lines above can be replaced by String format, that is usually more easy to read:

````
stringBuilder.append(String.format(" LIKE '%s%%'", value);
````

Or you can also append in each case the column instead of line 53, adding the value so it will be more easy to follow
 e.g.

````
stringBuilder.append(String.format("% LIKE '%s%%' ", columnName, value);
````

This advice is valid for all the case blocks.

Reply:
Thanks Jose, String.format indeed makes the code look a lot better. And I've moved columnName into String.format expression to make it more readable.